### PR TITLE
Refactor package

### PR DESCRIPTION
**Pull request by epels**
Mostly just moving stuff around: this refactor moves resources into their own packages, e.g.:

```go
resource, err := resource.Read(client, params)

// ...
```

Additionally, this PR introduces a new internal `messagebirdtest` package to DRY up test server/client across these packages.

### CODE REVIEW

File: mmsmessage/mms_message.go [hunk 1]
```diff
@@ -1,10 +1,13 @@
-package messagebird
+package mmsmessage
```
Comment (polyfloyd) [inline mmsmessage/mms_message.go]: What do you think about naming it just `mms`?

File: balance/balance.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package balance
```
Comment (polyfloyd) [inline balance/balance.go]: Balance is something really generic, what do you think about keeping it in the root package?

File: internal/messagebirdtest/test_client.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+// Write logs the provided buffer to the current test's error log.
+func (w testWriter) Write(p []byte) (int, error) {
+	w.t.Logf("%s", p)
+
+	return len(p), nil
+}
+
+// Client initializes a new MessageBird client that uses the
+func Client(t *testing.T) *messagebird.Client {
+	return client(t, "test_gshuPaZoeEG6ovbc8M79w0QyM")
```
Comment (polyfloyd) [inline internal/messagebirdtest/test_client.go]: Is it ok to commit this? Or should we rather set a secret in the CI env?

**Comment by epels** [comment]
Self review: rename structs like `MMSMessageParams` to `Params`, to avoid code like `mmsmessage.MMSMessageParam`.

CC @polyfloyd

File: message/message.go [hunk 4]
```diff
@@ -77,6 +80,49 @@ type messageRequest struct {
...
+
+	if err := c.Request(messageList, http.MethodGet, path+"?"+params.Encode(), nil); err != nil {
+		return nil, err
+	}
+
+	return messageList, nil
+}
+
+// Create creates a new message for one or more recipients.
+func Create(c *messagebird.Client, originator string, recipients []string, body string, msgParams *MessageParams) (*Message, error) {
```
Comment (polyfloyd) [inline message/message.go]: What do you think about calling this function `Send`?

File: mmsmessage/mms_message.go [hunk 5]
```diff
@@ -32,6 +35,37 @@ type MMSMessageParams struct {
...
+	mmsMessage := &MMSMessage{}
+	if err := c.Request(mmsMessage, http.MethodGet, path+"/"+id, nil); err != nil {
+		return nil, err
+	}
+
+	return mmsMessage, nil
+}
+
+// Create creates a new MMS message for one or more recipients.
+func Create(c *messagebird.Client, originator string, recipients []string, msgParams *MMSMessageParams) (*MMSMessage, error) {
```
Comment (polyfloyd) [inline mmsmessage/mms_message.go]: What do you think about calling this function `Send`?

File: mmsmessage/mms_message.go (hunk 1 above)
Comment (epels) [reply mmsmessage/mms_message.go]: Totally agree!

File: balance/balance.go (hunk 2 above)
Comment (epels) [reply balance/balance.go]: Moving this to the root package would create a circular dependency between `messagebird` and `messagebirdtest` (latter depends on `messagebird.Client`, and `balance_test.go` uses the `messagebirdtest` package for test server/client). Additionally, I guess this is a bit more consistent. Is it ok with you to keep it like this?

File: message/message.go (hunk 4 above)
Comment (epels) [reply message/message.go]: Hmm, within the wider context of resources, I'd say `Create` is a bit more suitable - I can definitely understand where you're coming from for message/mmsmessage, but it'd feel odd to have `message.Send()` but `lookup.Create()`. What's your opinion on this?

File: internal/messagebirdtest/test_client.go (hunk 3 above)
Comment (epels) [reply internal/messagebirdtest/test_client.go]: I'll look into how Travis facilitates this, thanks!

Comment (epels) [reply internal/messagebirdtest/test_client.go]: Hm, it seems it's easily done by encrypting it and setting it as an environment variable. The issue with this is that this won't work for PRs from forks, meaning the build would always fail for (external) contributors: https://docs.travis-ci.com/user/encryption-keys/

Comment (epels) [reply internal/messagebirdtest/test_client.go]: I have removed the test key from the test client for now, as it is not used at this point anyway.

**Pull request merged**